nodejs_day6/server.js

```diff
@@ -0,0 +1,160 @@
+import express from 'express';
+import dotenv from 'dotenv';
+import path from 'path';
+import flash from 'express-flash';
+//stroing session in memory psql database
+import session from 'express-session';
+import bcrypt from 'bcrypt';
+import passport from 'passport';
+import sequelize from './util/db';
+import register from './model/register';
+import initializePassport from './passport-config/passportConfig';
+
+const logout = require('express-passport-logout');
+
+initializePassport(passport);
+
+const app = express();
+dotenv.config();
+
+const port = process.env.PORT;
+
+sequelize
+  .authenticate()
+  .then(() => {
+    console.log('Connection has been established successfully.');
+  })
+  .catch((err) => {
+    console.error('Unable to connect to the database:', err);
+  });
+
+// do not create table if it already exists and use the existing table and data
+register
+  .sync({ force: false })
+  .then(() => {
+    console.log('Table created successfully');
+  })
+  .catch((err) => {
+    console.error('Error creating table:', err);
+  });
+
+app.use(express.urlencoded({ extended: false }));
+app.set('view engine', 'ejs');
+app.set('views', path.join(__dirname, 'views'));
+app.use(flash());
+app.use(
+  session({
+    // Key we want to keep secret which will encrypt all of our information
+    secret: process.env.SESSION_SECRET,
+    // Should we resave our session variables if nothing has changes which we dont
+    resave: false,
+    // Save empty value if there is no vaue which we do not want to do
+    saveUninitialized: false,
+    // Cookie options with expire in one minute
+  })
+);
+
+// Funtion inside passport which initializes passport
+app.use(passport.initialize());
+// Store our variables to be persisted across the whole session. Works with app.use(Session) above
+app.use(passport.session());
+
+app.get('/', (req, res) => {
+  res.render('index');
+});
+
+function checkAuthenticated(req, res, next) {
+  if (req.isAuthenticated()) {
+    return res.redirect('/users/dashboard');
+  }
+  next();
+}
+
+function checkNotAuthenticated(req, res, next) {
+  if (req.isAuthenticated()) {
+    return next();
+  }
+  res.redirect('/users/login');
+}
+
+app.get('/users/register', checkAuthenticated, (req, res) => {
+  res.render('register.ejs');
+});
+
+app.get('/users/login', checkAuthenticated, (req, res) => {
+  //flash sets a messages variable. passport sets the error message
+  console.log(req.session.flash.error);
+  res.render('login.ejs');
+});
+
+app.get('/users/dashboard', checkNotAuthenticated, async (req, res) => {
```

Comment:
if using async await perfer using async await in whole application as currently in some place there it then catch implemented
Also prefer try catch with async and await for error handling